code/obj/item/gun/kinetic.dm

```diff
@@ -623,23 +623,67 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
 	name = "Riot Shotgun"
 	desc = "A police-issue shotgun meant for suppressing riots."
 	icon = 'icons/obj/48x32.dmi'
-	icon_state = "shotty"
+	icon_state = "shotty-empty"
 	item_state = "shotty"
 	force = 15.0
 	contraband = 5
 	caliber = 0.72
 	max_ammo_capacity = 8
-	auto_eject = 1
+	auto_eject = 0
 	can_dual_wield = 0
 	two_handed = 1
 	has_empty_state = 1
 	gildable = 1
+	var/racked_slide = FALSE
+
 
 	New()
 		ammo = new/obj/item/ammo/bullets/abg
 		set_current_projectile(new/datum/projectile/bullet/abg)
 		..()
 
+	update_icon()
+		. = ..()
+		src.icon_state = "shotty" + (gilded ? "-golden" : "") + (racked_slide ? "" : "-empty" )
+
+	canshoot()
+		if (src.racked_slide)
+			return ..()
+		else
+			boutput(usr, "<span class='notice'>You need to rack the slide before you can fire!</span>") //Pain
+			return 0
+
+	shoot(var/target,var/start ,var/mob/user)
+		..()
+		src.racked_slide = FALSE
+		if (src.ammo.amount_left == 0) // change icon_state to empty if 0 shells left
+			src.update_icon()
+		src.casings_to_eject = 0 //This probably works
+
+	attack_self(mob/user as mob)
+		..()
+		if (!src.racked_slide) //Are we racked?
+			if (src.ammo.amount_left == 0)
+				boutput(user, "<span class ='notice'>You are out of shells!</span>")
+				src.icon_state = "shotty" + (src.gilded ? "-golden-empty" : "-empty")
+			else
+				src.racked_slide = TRUE
+				if (src.icon_state == "shotty[src.gilded ? "-golden" : ""]") //"animated" racking
+					src.icon_state = "shotty[src.gilded ? "-golden-empty" : "-empty"]"
+					animate(src, time = 0.2 SECONDS) //thank you pali for telling me about animate
+					animate(icon_state = "shotty[gilded ? "-golden" : ""]")
+				else
+					src.icon_state = "shotty[gilded ? "-golden" : ""]" // Slide already open? Just close the slide
```

Comment:
why are you doing this stuff manually instead of update_icon

Reply:
Brain fart. Fixed in commit